src/promptflow/promptflow/_telemetry/activity.py

```diff
@@ -51,11 +51,16 @@ def log_activity(
     :type custom_dimensions: dict
     :return: None
     """
+    from promptflow._sdk._utils import setup_user_agent_to_operation_context
+
+    user_agent = setup_user_agent_to_operation_context()
```

Comment:
maybe change to get_xx

Reply:
OperationContext.get_instance().get_user_agent()?

Reply:
I suggest we keep only the `append_user_agent` and `get_user_agent `functions. Adding too many functions for modifying the agent might lead to mistake by others use. Therefore, if necessary, should the function `update_user_agent_from_env_var() `be merged into `append_user_agent`?